crates/matrix-sdk/src/event_cache/linked_chunk/updates.rs

```diff
@@ -61,23 +61,31 @@ pub enum Update<Item, Gap> {
     /// A chunk has been removed.
     RemoveChunk(ChunkIdentifier),
 
-    /// Items are inserted inside a chunk of kind Items.
-    InsertItems {
-        /// [`Position`] of the items.
-        at: Position,
+    /// Items are pushed inside a chunk of kind Items.
+    PushItems {
+        /// The [`Position`] of the items.
+        ///
+        /// This value is given to prevent the need for position computations by
+        /// the update readers. Items are pushed, so the positions should be
+        /// incrementally computed from the previous items, which requires the
+        /// reading of the last previous item. With `position_hint`, the update
+        /// readers no longer need to do so.
+        position_hint: Position,
 
         /// The items.
         items: Vec<Item>,
     },
 
-    /// A chunk of kind Items has been truncated.
-    TruncateItems {
-        /// The identifier of the chunk.
-        chunk: ChunkIdentifier,
-
-        /// The new length of the chunk.
-        length: usize,
+    /// The last items of a chunk have been detached, i.e. the chunk has been
+    /// truncated.
+    DetachLastItems {
```

Comment:
Not sure this name is very telling, after reading it in context. Could we rename this to `SplitOff`, maybe?

Reply:
`Detach…` was playing well with `Reattach…` 😇.